### ISSUE TITLE
IPython console fails to start when using custom interpreter because of lack of permissions in temp dir

### ISSUE THREAD

**Issue by udsharma1506** [spyder-ide/spyder#6481]
## Description

**What steps will reproduce the problem?**

1. Console -> Open an Ipython Console 
2. When the new console opens up, it crashed

**What is the expected output? What do you see instead?**
It should have opened the new console for me.

**Please provide any additional information below**

  File "/Users/udsharma/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py", line 976, in create_new_client
    interpreter=pyexec)
  File "/Users/udsharma/anaconda2/lib/python2.7/site-packages/spyder/utils/programs.py", line 405, in is_module_installed
    fd, script = tempfile.mkstemp(suffix='.py', dir=TEMPDIR)
  File "/Users/udsharma/anaconda2/lib/python2.7/tempfile.py", line 314, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags)
  File "/Users/udsharma/anaconda2/lib/python2.7/tempfile.py", line 244, in _mkstemp_inner
    fd = _os.open(file, flags, 0600)
OSError: [Errno 2] No such file or directory: '/var/folders/y9/fn0t45pj4wv7jy59xwx6gzkm0000gn/T/spyder-udsharma/tmpv3jxd_.py'

## Version and main components

* Spyder Version: 3.2.3 
* Python Version: 2.7.13
* Qt Versions: 5.6.2, PyQt5 5.6 on Darwin

## Dependencies
```
pyflakes >=0.5.0  :  1.5.0 (OK)
pycodestyle >=2.3 :  2.3.1 (OK)
pygments >=2.0    :  2.2.0 (OK)
pandas >=0.13.1   :  0.19.2 (OK)
numpy >=1.7       :  1.13.3 (OK)
sphinx >=0.6.6    :  1.6.3 (OK)
rope >=0.9.4      :  0.10.5 (OK)
jedi >=0.9.0      :  0.10.2 (OK)
psutil >=0.3      :  5.2.2 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
sympy >=0.7.3     :  1.1.1 (OK)
cython >=0.21     :  0.26.1 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
IPython >=4.0;<6.0:  5.4.1 (OK)
pylint >=0.25     :  1.7.2 (OK)

```

**Comment by ccordoba12** [comment]
> OSError: [Errno 2] No such file or directory: '/var/folders/y9/fn0t45pj4wv7jy59xwx6gzkm0000gn/T/spyder-udsharma/tmpv3jxd.py'

This error means you don't have permissions to write in your own temporary directory. Without that, Spyder can't use a different interpreter from the default one, because it needs to do some validations using a temporary file.

Solution: please make sure that your assigned temporary is writable.

**Comment by ccordoba12** [comment]
Note to self: When selecting a custom interpreter, we need to run a validation to see if TEMPDIR is writable and if not we need to inform users that they can't use it.

**Comment by udsharma1506** [comment]
Solution: please make sure that your assigned temporary is writable.

How can I do that, can you help me with the steps?

**Comment by ccordoba12** [comment]
Running in a terminal (xterm, konsole, etc):

    chmod u+w /var/folders/y9/fn0t45pj4wv7jy59xwx6gzkm0000gn/T/spyder-udsharma/

should do it.

**Comment by ccordoba12** [comment]
Closing as duplicate of #7800, which was already fixed and it'll be part of our 3.3.2 release. In the meantime, please avoid using two Spyder instances and only use one.
